controllers/datadogagent/feature/dogstatsd/feature.go

```diff
@@ -0,0 +1,198 @@
+	dogstatsd := dda.Spec.Features.Dogstatsd
+	if apiutils.BoolValue(dogstatsd.HostPortConfig.Enabled) {
+		f.hostPortEnabled = true
+	}
+	if dogstatsd.HostPortConfig.Port != nil {
+		f.hostPortHostPort = *dogstatsd.HostPortConfig.Port
+	}
+	if apiutils.BoolValue(dogstatsd.UnixDomainSocketConfig.Enabled) {
+		f.udsEnabled = true
+	}
+	f.udsHostFilepath = *dogstatsd.UnixDomainSocketConfig.Path
+	if apiutils.BoolValue(dogstatsd.OriginDetectionEnabled) {
+		f.originDetectionEnabled = true
+	}
+	if dogstatsd.MapperProfiles != nil {
+		f.mapperProfiles = v2alpha1.ConvertCustomConfig(dogstatsd.MapperProfiles)
+	}
+	reqComp = feature.RequiredComponents{
+		Agent: feature.RequiredComponent{
+			Containers: []apicommonv1.AgentContainerName{
+				apicommonv1.CoreAgentContainerName,
+			},
+		},
+	}
+	return reqComp
+}
+
+// ConfigureV1 use to configure the feature from a v1alpha1.DatadogAgent instance.
+func (f *dogStatsDFeature) ConfigureV1(dda *v1alpha1.DatadogAgent) (reqComp feature.RequiredComponents) {
+	config := dda.Spec.Agent.Config
+	f.hostPortEnabled = true
+	if config.HostPort != nil {
+		f.hostPortHostPort = *config.HostPort
+	}
+	if apiutils.BoolValue(config.Dogstatsd.UnixDomainSocket.Enabled) {
+		f.udsEnabled = true
+	}
+	if config.Dogstatsd.UnixDomainSocket.HostFilepath != nil {
+		f.udsHostFilepath = *config.Dogstatsd.UnixDomainSocket.HostFilepath
+	}
+	if apiutils.BoolValue(config.Dogstatsd.DogstatsdOriginDetection) {
+		f.originDetectionEnabled = true
+	}
+	if config.Dogstatsd.MapperProfiles != nil {
+		f.mapperProfiles = v1alpha1.ConvertCustomConfig(config.Dogstatsd.MapperProfiles)
+	}
+	reqComp = feature.RequiredComponents{
+		Agent: feature.RequiredComponent{
+			Containers: []apicommonv1.AgentContainerName{
+				apicommonv1.CoreAgentContainerName,
+			},
+		},
+	}
+	return reqComp
+}
+
+// ManageDependencies allows a feature to manage its dependencies.
+// Feature's dependencies should be added in the store.
+func (f *dogStatsDFeature) ManageDependencies(managers feature.ResourceManagers) error {
+	return nil
+}
+
+// ManageClusterAgent allows a feature to configure the ClusterAgent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *dogStatsDFeature) ManageClusterAgent(managers feature.PodTemplateManagers) error {
+	return nil
+}
+
+// ManageNodeAgent allows a feature to configure the Node Agent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *dogStatsDFeature) ManageNodeAgent(managers feature.PodTemplateManagers) error {
+	// udp
+	if f.hostPortEnabled {
+		if f.hostPortHostPort != 0 {
+			managers.Port().AddPortToContainer(apicommonv1.CoreAgentContainerName, &corev1.ContainerPort{
+				Name:          apicommon.DogStatsDHostPortName,
+				HostPort:      f.hostPortHostPort,
+				ContainerPort: apicommon.DogStatsDHostPortHostPort,
```

Comment:
not really familiar with dogstatsd config; should this be an option in `Feature.Dogstatsd`?

Reply:
I misunderstood how the hostPort would be set in v1alpha1 vs v2alpha1. In v1alpha1, the hostPort isn’t set unless specified and I don’t think there is a default value. If I don’t specify anything in a DDA, only the containerPort is set for the agent: [datadog-operator/utils.go at 88fe84904b05d8e7c1793b41a5b0f8cd5f7cf655 · DataDog/datadog-operator · GitHub](https://github.com/DataDog/datadog-operator/blob/88fe84904b05d8e7c1793b41a5b0f8cd5f7cf655/controllers/datadogagent/utils.go#L298)
```
    ports:
    - containerPort: 8125
      name: dogstatsdport
      protocol: UDP
```
If I specify a hostPort in the DDA config, that hostPort is added: [datadog-operator/utils.go at 88fe84904b05d8e7c1793b41a5b0f8cd5f7cf655 · DataDog/datadog-operator · GitHub](https://github.com/DataDog/datadog-operator/blob/88fe84904b05d8e7c1793b41a5b0f8cd5f7cf655/controllers/datadogagent/utils.go#L304)
```
    ports:
    - containerPort: 8125
      hostPort: 7890
      name: dogstatsdport
      protocol: UDP
```

In v2alpha, it looks like when the hostPort is enabled, the hostPort port value will default to 8125 if not specified: [datadog-operator/datadogagent_default.go at main · DataDog/datadog-operator · GitHub](https://github.com/DataDog/datadog-operator/blob/main/apis/datadoghq/v2alpha1/datadogagent_default.go#L176)